pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -372,7 +382,9 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
                 lr_schedulers.append(config.scheduler.state_dict())
             checkpoint["lr_schedulers"] = lr_schedulers
 
-            self.trainer.precision_plugin.on_save_checkpoint(checkpoint)
+            # precision plugin
+            prec_plugin = self.trainer.precision_plugin
+            checkpoint[prec_plugin.__class__.__name__] = self.trainer.precision_plugin.state_dict()
```

Comment:
note: now if the precision plugin/module ever changes, loading old checkpoints will fail

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/86b177ebe5427725b35fde1a8808a7b59b8a277a/pytorch_lightning/callbacks/base.py#L43 

let's use qualname here too for consistency

Reply:
> note: now if the precision plugin/module ever changes, loading old checkpoints will fail

if I understand this comment correctly - for compatibility with old checkpoint keys, added an `isinstance` check for the precision plugin before calling `load_state_dict(ckpt["old_precplugin_key"])`